File: js/example/sub-client.js
var riffle = require('jsriffle');

riffle.SetFabricLocal();
riffle.SetLogLevelDebug();

var app = riffle.Domain("xs.demo.test");
var backend = app.subdomain("backend");
var client = app.subdomain("client");

client.onJoin = function() {

    // Example Pub/Sub Basic - a very basic pub/sub example
    backend.publish("basicSub", "Hello");
    // End Example Pub/Sub Basic
    
    // Example Pub/Sub Basic Two - a basic pub/sub example
    backend.publish("basicSubTwo", "Hello", 3);
    // End Example Pub/Sub Basic Two

};

client.join()

File: python/example/sub-client.py

import riffle

riffle.SetFabricLocal()
riffle.SetLogLevelDebug()


class Send(riffle.Domain):

    def onJoin(self):

        # Example Pub/Sub Basic - This is a basic pub/sub
        backend.publish("basicSub", "Hello")
        # End Example Pub/Sub Basic

        # Example Pub/Sub Objects - This tests sending an object using pub/sub
        class Stuff(riffle.ModelObject):
            name = "This guy"

        s = Stuff()
        backend.publish("objectSub", s)
        # End Example Pub/Sub Objects

        # self.leave()


app = riffle.Domain("xs.demo.test")

if __name__ == '__main__':
    Send("example", superdomain=app).join()
    exit()
